homeassistant/components/google_assistant_sdk/helpers.py

```diff
@@ -113,6 +113,33 @@ def default_language_code(hass: HomeAssistant) -> str:
     return DEFAULT_LANGUAGE_CODES.get(hass.config.language, "en-US")
 
 
+def best_matching_language_code(
```

Comment:
Have you looked at the utility functions in homeassistant/util/language.py?

Reply:
Yes, I’ve had a look at those functions… but they are not really helping. 
The main problem targeted by this merge request is the following:

This integration allows the user to select one of the (google-) supported languages.
But when used as part of an assist pipeline, the selected language (e.g. `en-AU`) of the integration is ignored, but the assist language (just `en`) is used. But just `en` is not supported by google. But which dialect should be used? The default (`en-US`) or the one configured in the integration (e.g. `en-AU`)? Both are supported by google. I‘d suggest to prioritize the selected language from the integration. Of course it could happen that the user configures a completely different language at the integration (e.g. Chinese) than in the assist pipeline (e.g. English). For that we needed a decision tree. Which language shall be preferred? I think the unit test shows the decision tree quite good. 

The utils you mentioned are good to use for intersecting lists or finding a general weak match, but won‘t help on that decision tree. Additionally the `Dialect` class seems to be useful, but doesn’t support multiple dialects to find the best match. Also those scores are not comparable, since it’s just 4 different results not showing that one English dialect is better than the other based on the place where it was configured.